gamla/functional.py

```diff
@@ -1025,3 +1000,24 @@ def ends_with(seq: Iterable):
         return True
 
     return ends_with
+
+
+def intersect(collections: Collection[Collection]) -> Iterable:
+    """Intersects a group of collections.
+
+    >>> tuple(
+            gamla.intersect(
+                [
+                    [1, 2, 3, 4],
+                    [4, 5],
+                    [4, 2],
+                ]
+            )
+        )
+    (4,)
+
+    """
+    first_collection, *rest = collections
+    for x in first_collection:
+        if all(x in container for container in rest):
```

Comment:
this could be quadratic no ? maybe turn the collections into sets ? or at least add a comment in the function's docs...

Reply:
It can be quadratic if the user doesn't give a container with O(1) containment check.
however if I were to turn the collections into sets I cause an O(n) for users that use the generator ability and can get away with O(1).
I'll add a comment.